src/contracts/core/DelegationManager.sol

```diff
@@ -341,239 +406,245 @@ contract DelegationManager is
-        } else {
-            // Award shares back in StrategyManager/EigenPodManager.
-            // If withdrawer is delegated, increase the shares delegated to the operator.
-            address currentOperator = delegatedTo[msg.sender];
-            for (uint256 i = 0; i < withdrawal.strategies.length;) {
-                require(
-                    withdrawal.startBlock + strategyWithdrawalDelayBlocks[withdrawal.strategies[i]] <= block.number,
-                    WithdrawalDelayNotElapsed()
-                );
-
-                /**
-                 * When awarding podOwnerShares in EigenPodManager, we need to be sure to only give them back to the original podOwner.
-                 * Other strategy shares can + will be awarded to the withdrawer.
-                 */
-                if (withdrawal.strategies[i] == beaconChainETHStrategy) {
-                    address staker = withdrawal.staker;
-                    /**
-                     * Update shares amount depending upon the returned value.
-                     * The return value will be lower than the input value in the case where the staker has an existing share deficit
-                     */
-                    uint256 increaseInDelegateableShares =
-                        eigenPodManager.addShares({podOwner: staker, shares: withdrawal.shares[i]});
-                    address podOwnerOperator = delegatedTo[staker];
-                    // Similar to `isDelegated` logic
-                    if (podOwnerOperator != address(0)) {
-                        _increaseOperatorShares({
-                            operator: podOwnerOperator,
-                            // the 'staker' here is the address receiving new shares
-                            staker: staker,
-                            strategy: withdrawal.strategies[i],
-                            shares: increaseInDelegateableShares
-                        });
-                    }
-                } else {
-                    strategyManager.addShares(msg.sender, tokens[i], withdrawal.strategies[i], withdrawal.shares[i]);
-                    // Similar to `isDelegated` logic
-                    if (currentOperator != address(0)) {
-                        _increaseOperatorShares({
-                            operator: currentOperator,
-                            // the 'staker' here is the address receiving new shares
-                            staker: msg.sender,
-                            strategy: withdrawal.strategies[i],
-                            shares: withdrawal.shares[i]
-                        });
-                    }
-                }
-                unchecked {
-                    ++i;
-                }
             }
         }
 
-        emit WithdrawalCompleted(withdrawalRoot);
+        _stakerQueuedWithdrawalRoots[withdrawal.staker].remove(withdrawalRoot);
+
+        delete queuedWithdrawals[withdrawalRoot];
+        delete pendingWithdrawals[withdrawalRoot];
```

Comment:
possible bug: moving this down in the function breaks the CEI pattern, maybe opening the path to reentrancy via a malicious token. Possible attack would look like (1) initiate withdrawal containing multiple tokens (2) once callstack makes it to malicious token, initiate the same withdrawal a second time via reentrant call (3) get 2x tokens

I highly recommend moving these deletions as far up in this function as possible, particularly before any calls external to the protocol may occur.

It's super possible there's no actual threat here, but this reordering should be purely safer either way.

Reply:
We do have `nonReentrant` on all completeQueuedWithdrawal interfaces so this shouldn't be an issue but I do agree on following CEI more closely.